### PR TITLE
test: add comprehensive unit test coverage

### PR DESCRIPTION
**Pull request by nicklamont**
## Summary
- Add unit tests for formatting utilities (`convert_formatting`, `parse_slack_blocks`, `_parse_rich_text_elements`)
- Add unit tests for `slack_ts_to_rfc3339` timestamp conversion
- Add unit tests for `generate_user_map` (domain override, bot ignoring, overrides, missing emails)
- Extend existing config tests with `create_default_config`, edge cases, invalid YAML, nonexistent paths
- Add unit tests for `channel_has_external_users` and space management named constants
- Add unit tests for `track_message_stats` (reactions, files, dry run, bot filtering)
- Add unit tests for migrator initialization (email validation, `_is_external_user`, domain extraction)

Grows test suite from 3 to 93 tests, all pure unit tests with no external dependencies.

## Test plan
- [x] `python -m pytest tests/ -v` — 93 passed
- [x] `ruff check .` passes
- [x] `python -c "import slack_migrator"` clean import

### CODE REVIEW

File: tests/unit/test_formatting.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        result = convert_formatting("<https://example.com>", {})
+        assert result == "https://example.com"
+
+    def test_special_mention(self):
+        result = convert_formatting("<!here> <!channel|channel>", {})
+        assert result == "@here @channel"
+
+    def test_emoji_conversion(self):
+        result = convert_formatting("great :thumbsup:", {})
+        assert ":thumbsup:" not in result or "👍" in result
```
[the remark this answers is not in the document]
Comment (nicklamont) [reply tests/unit/test_formatting.py]: Agreed — strengthened to `assert "👍" in result` which directly verifies the emoji library converted the alias. Fixed in 5f6e3af.
